services/structureddate/structureddate/src/main/java/org/collectionspace/services/structureddate/antlr/ANTLRStructuredDateEvaluator.java

```diff
@@ -293,6 +314,26 @@ public void exitHyphenatedRange(HyphenatedRangeContext ctx) {
 			((DeferredDate) latestEndDate).resolveDate();
 		}
 
+		// Check if they are reversed
+		if (latestEndDate.getEra() == Era.BCE && earliestStartDate.getEra() == Era.BCE) {
+			if (earliestStartDate.getYear() < latestEndDate.getYear()) {
+				Date tmp = earliestStartDate;
+				earliestStartDate = latestEndDate;
+				latestEndDate = tmp;
+
+				// This is probably bad technique, but right now, if the month and days were calculated
+				// Chances are that the EarliestStart date has the LatestStart date
+				// And the latestEnd date has the earliestEnd date. So we set them accordingly.
+				if ((earliestStartDate.getMonth() == 12 && earliestStartDate.getDay() == 31) &&
+					(latestEndDate.getMonth() == 1 && latestEndDate.getDay() == 1)) {
+						earliestStartDate.setMonth(1);
+						earliestStartDate.setDay(1);
+						latestEndDate.setMonth(12);
+						latestEndDate.setDay(31);
+				}
+			}
+		}
+
```

Comment:
I would do this at the very end, at the top of the tree (exitDisplayDate). There's no reason to be limited to BC dates. At the end, check If the earliest is after the latest, and if so, swap them.